event_model/__init__.py

```diff
@@ -465,19 +480,37 @@ def event_page(self, doc):
         # efficient than unpacking the page in to Events, filling them, and the
         # re-packing a new page. But that seems tricky in general since the
         # page may be implemented as a DataFrame or dict, etc.
-
-        event = self.event  # Avoid attribute lookup in hot loop.
-        filled_events = []
-
-        for event_doc in unpack_event_page(doc):
-            filled_events.append(event(event_doc))
-        new_event_page = pack_event_page(*filled_events)
+        filled_doc = self.fill_event_page(doc, include=self.include,
+                                          exclude=self.exclude)
         # Modify original doc in place, as we do with 'event'.
-        doc['data'] = new_event_page['data']
-        doc['filled'] = new_event_page['filled']
-        return doc
+        if self._inplace:
+            doc['data'] = filled_doc['data']
+            doc['filled'] = filled_doc['filled']
+            return doc
+        else:
+            return filled_doc
 
     def event(self, doc):
+        filled_doc = self.fill_event(doc, include=self.include,
+                                     exclude=self.exclude)
+        if self._inplace:
```

Comment:
Would it make more sense to do this logic it `fill_event` (and `fill_event_page`) so that if the user wants inplace we can skip the deepcopy all together an so that `event` and `fill_event` follow the same `inplace` rules?